core/src/test/java/io/cucumber/core/stepexpression/StepExpressionFactoryTest.java

```diff
@@ -103,13 +104,39 @@ void unknown_target_type_does_no_transform_data_table() {
     }
 
     @Test
-    void unknown_target_type_does_not_transform_doc_string() {
+    void unknown_target_type_transform_doc_string_to_string_object() {
         String docString = "A rather long and boring string of documentation";
+        String contentType = null;
         StepExpression expression = new StepExpressionFactory(registry).createExpression("Given some stuff:", UNKNOWN_TYPE);
-        List<Argument> match = expression.match("Given some stuff:", docString);
+        List<Argument> match = expression.match("Given some stuff:", docString, contentType);
         assertThat(match.get(0).getValue(), is(equalTo(docString)));
     }
 
+    @Test
+    void docstring_expression_transform_doc_string_to_json_node() {
+        String docString = "{\"hello\": \"world\"}";
+        String contentType = "json";
+        registry.defineDocStringType(new DocStringType(JsonNode.class, contentType, (String s) -> objectMapper.convertValue(docString, JsonNode.class)));
+
+        StepExpression expression = new StepExpressionFactory(registry).createExpression("Given some stuff:", JsonNode.class);
+        List<Argument> match = expression.match("Given some stuff:", docString, contentType);
+        JsonNode node = (JsonNode) match.get(0).getValue();
+        assertThat(node.asText(), equalTo(docString));
+    }
+
+    @Test
+    void docstring_expression_transform_anonymous_doc_string_type() {
+        String docString = "A rather long and boring string of documentation";
+        String contentType = null;
```

Comment:
This should result in an error. The content type must always be a string. And the string may not be empty. With the exception of the nameless doc string type we predefined.